### PR TITLE
[5.3] Adding two new method next() and previous() to Collection

### PR DESCRIPTION
**Pull request by fernandobandeira**
Not sure if this will be useful for anyone else.

Added two methods to  the Collection class, next() and previous().

This works with any sort of order and keys, as far as i could test, however i guess that we could come up with a better idea for the previous method.

Also this is useful when you want to loop the next and previous links.

### CODE REVIEW

**Comment by fernandobandeira** [comment]
Gonna fix the Style CI, not sure why the travis failed on 7.1 only? ``` Memory exhausted ```

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -606,15 +606,15 @@ public function previous($currentItem, $loop = false)
...
     {
         $current = $this->search($currentItem, true);
-        $current = $this->keys()->search($current, true);	
+        $current = $this->keys()->search($current, true);
 
         $next = $this->slice($current, 2);
 
         if ($next->count() == 2) {
             return $next->last();
         } elseif ($loop) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: just if would be fine here

File: src/Illuminate/Support/Collection.php [hunk 2]
```diff
@@ -573,6 +573,53 @@ public function last(callable $callback = null, $default = null)
...
+     */
+    public function previous($currentItem, $loop = false)
+    {
+        $reversedCollection = $this->reverse();
+        $current = $reversedCollection->search($currentItem, true);
+        $current = $reversedCollection->keys()->search($current, true);
+
+        $previous = $reversedCollection->slice($current, 2);
+
+        if ($previous->count() == 2) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: === please

**Comment by GrahamCampbell** [comment]
Don't worry about PHP 7.1. There must be a bug in PHP itself since it's segfaulting. Not our fault. :)

**Comment by taylorotwell** [comment]
No plans on adding this to core.

**Pull request closed without merging**